File: data/src/main/java/org/oppia/data/backends/gae/model/Story.java
package org.oppia.data.backends.gae.model;

import com.squareup.moshi.Json;

import java.util.List;

public class Story {

  /*
   * Parent Hierarchy: Story
   */

  //TODO: ImmutableList<> and List<> error
  //ImmutableList<> does not work but List<> works

  @Json(name = "story_title")
  public String storyTitle;

  @Json(name = "story_description")
  public String storyDescription;

  @Json(name = "story_nodes")
  public List<StoryNode> storyNodes;

}

File: data/src/main/java/org/oppia/data/backends/gae/model/Story.kt
package org.oppia.data.backends.gae.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Story(

  /*
   * Ignore below params
   * is_admin, iframed, is_moderator,
   * is_super_admin, state_classifier_mapping,
   * preferred_audio_language_code, can_edit,
   * is_topic_manager, additional_angular_modules
   * auto_tts_enabled
   */

  @Json(name = "story_title") val storyTitle: String?,
  @Json(name = "story_description") val storyDescription: String?,
  @Json(name = "story_nodes") val storyNodes: List<StoryNode>?

)
